aiohttp/client_reqrep.py

```diff
@@ -190,623 +193,352 @@ class ConnectionKey(NamedTuple):
     proxy_headers_hash: Optional[int]  # hash(CIMultiDict)
 
 
-class ClientRequest:
-    GET_METHODS = {
-        hdrs.METH_GET,
-        hdrs.METH_HEAD,
-        hdrs.METH_OPTIONS,
-        hdrs.METH_TRACE,
-    }
-    POST_METHODS = {hdrs.METH_PATCH, hdrs.METH_POST, hdrs.METH_PUT}
-    ALL_METHODS = GET_METHODS.union(POST_METHODS).union({hdrs.METH_DELETE})
-
-    DEFAULT_HEADERS = {
-        hdrs.ACCEPT: "*/*",
-        hdrs.ACCEPT_ENCODING: _gen_default_accept_encoding(),
-    }
-
-    # Type of body depends on PAYLOAD_REGISTRY, which is dynamic.
-    body: Any = b""
-    auth = None
-    response = None
+class ClientResponse(HeadersMixin):
```

Comment:
There are 0 changes to ClientResponse, I've just moved it up so I can reference it in ClientRequest.

Reply:
I don't that makes a difference? I've just added a reference to ClientResponse (the class, not an instance) as a class attribute in ClientRequest, so need it defined first.